lab5/utils/FileManager.java

```diff
@@ -0,0 +1,81 @@
+package org.example.utils;
+
+
+import org.example.collections.City;
+import org.example.collections.CityCollection;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import java.io.*;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+
+public class FileManager {
+    private final CollectionManager collectionManager;
+
+    public FileManager(CollectionManager collectionManager) {
+        this.collectionManager = collectionManager;
+    }
+
+    public void saveToXml(String filePath){
+        try (PrintWriter writer = new PrintWriter(new FileWriter(filePath))) {
+            // Create JAXB context for the CollectionManager class
+            JAXBContext context = JAXBContext.newInstance(CityCollection.class);
+            Marshaller marshaller = context.createMarshaller();
+
+            // Format the output for better readability
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+
+            // Marshal the CityCollection object to XML and save it to the file
+            marshaller.marshal(collectionManager.getCityCollection(), writer);
+
+            System.out.println("Collection saved to " + filePath);
+        } catch (JAXBException e) {
+            System.out.println("Error saving collection: " + e.getMessage());
+        } catch (IOException e) {
+            System.out.println("Error writing to file: " + e.getMessage());
+        }
+    }
+
+    public void loadCollectionFromXml(String filePath) {

```

Comment:
No need to update collection manager here, it's not the responsibility of this class

It should just return city collection and someone else will create a collection manager with it